src/components/AddDestinationDialog/AddDestinationDialog.tsx

```diff
@@ -0,0 +1,104 @@
+import { Metadata } from '@bsv/spv-wallet-js-client';
+import { QuestionMarkCircleIcon as Question } from '@heroicons/react/24/outline';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { CirclePlus } from 'lucide-react';
+import React, { useState } from 'react';
+
+import { toast } from 'sonner';
+
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+  Label,
+  Textarea,
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from '@/components';
+import { useSpvWalletClient } from '@/contexts';
+import { errorWrapper } from '@/utils';
+
+export interface AddDestinationDialogProps {
+  className?: string;
+}
+
+export const AddDestinationDialog = ({ className }: AddDestinationDialogProps) => {
+  const [isAddDialogOpen, setIsAddDialogOpen] = useState(false);
+  const [metadata, setMetadata] = useState(JSON.stringify({}));
+
+  const { spvWalletClient } = useSpvWalletClient();
+  const queryClient = useQueryClient();
+
+  const mutation = useMutation({
+    mutationFn: async (metadata: Metadata) => {
+      // At this point, spvWalletClient is defined; using non-null assertion.
+      return await spvWalletClient!.NewDestination(metadata);
+    },
+    onSuccess: () => {
+      return queryClient.invalidateQueries({
+        queryKey: ['destinations'],
+      });
+    },
+  });
+
+  const handleAdd = async () => {
+    try {
+      const metadataParsed = JSON.parse(metadata) as Metadata;
+      mutation.mutate(metadataParsed);
+      setIsAddDialogOpen(false);
+      toast.success('Destination added');
+    } catch (error) {
+      errorWrapper(error);
+      toast.error('Failed to add destination');
+    }
+  };
+  const handleMetadataChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setMetadata(event.target.value);
+  };
+
+  const handleDialogOpen = () => {
+    setIsAddDialogOpen((prev) => !prev);
+  };
```

Comment:
The name of the function doesn't fit with the implementation.
Change the name to sth like: `handleDialogToggle` or change the implementation to `setIsAddDialogOpen(true)`.

Reply:
Changed